bpf/lib/conntrack.h

```diff
@@ -51,9 +52,13 @@ static inline void __inline__ __ct_update_timeout(struct ct_entry *entry,
 #endif
 }
 
-static inline void __inline__ ct_update_timeout(struct ct_entry *entry)
+static inline void __inline__ ct_update_timeout(struct ct_entry *entry, bool syn)
 {
-	__ct_update_timeout(entry, CT_DEFAULT_LIFETIME);
+	entry->seen_non_syn |= !syn;
+	if (entry->seen_non_syn)
+		__ct_update_timeout(entry, CT_DEFAULT_LIFETIME);
+	else
+		__ct_update_timeout(entry, CT_SYN_TIMEOUT);
```

Comment:
`__ct_update_timeout()` is protected by `NEEDS_TIMEOUT` which is undefined if we are running with LRU. The expiration needs to run with LRU as well.

Reply:
Good point, I added another commit to enable these timeouts in LRU case and to enable garbage collection in userspace for conntrack maps backed by LRU map type.